rocky/rocky/views/scheduler.py

```diff
@@ -197,9 +197,13 @@ def get_schedule_details(self, schedule_id: str) -> ScheduleResponse:
         except SchedulerError as error:
             return messages.error(self.request, error.message)
 
-    def get_schedule_with_filters(self, filters: dict[str, list[dict[str, str]]]) -> ScheduleResponse:
+    def get_schedule_with_filters(self, filters: dict[str, list[dict[str, str]]]) -> ScheduleResponse | None:
         try:
-            return self.scheduler_client.post_schedule_search(filters).results[0]
+            schedule = self.scheduler_client.post_schedule_search(filters)
```

Comment:
Double check to make sure the filters / query send to the scheduler actually include the organization. Otherwise we are not filtering on organization, and a malicious user could get any schedule by guessing the schedule ID